go.mod

```diff
@@ -12,6 +12,7 @@ require (
 	golang.org/x/crypto v0.27.0
 	golang.org/x/exp v0.0.0-20240909161429-701f63a606c0
 	golang.org/x/sync v0.8.0
+	golang.org/x/time v0.3.0
```

Comment:
We want this for ratelimiting?

Reply:
Yeah, this is exactly how it's done for v1.5, let's keep it as it